### PR TITLE
Static Credential Store

### PR DESCRIPTION
**Pull request by DanielleMiu**
adds static credential store to the terraform boundary provider

including 
- [x] automated tests
  - [x] automated tests _passing_
- [x] manual testing
- [x] examples

## testing details
- added below tf snippet, confirmed static credential store created
- removed below tf snipped, confirmed static credential store destroyed

```
resource "boundary_credential_store_static" "example" {
  name        = "example_static_credential_store"
  description = "this is a static credential store being created for testing purposes only."
  scope_id    = boundary_scope.project.id
}
```
```
boundary credential-stores list -scope-id <scope_id>
```

### CODE REVIEW

File: internal/provider/provider.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func New() *schema.Provider {
 			"boundary_auth_method_oidc":         resourceAuthMethodOidc(),
 			"boundary_credential_library_vault": resourceCredentialLibraryVault(),
 			"boundary_credential_store_vault":   resourceCredentialStoreVault(),
+			"boundary_credential_store_static":  resourceStaticCredentialStore(),
```
Comment (johanbrandhorst) [inline internal/provider/provider.go:73]: Nit, can we name the function similarly to the others too?
```suggestion
			"boundary_credential_store_static":  resourceCredentialStoreStatic(),
```

File: internal/provider/resource_credential_store_static_test.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	md := testProvider.Meta().(*metaData)
+	c := credentialstores.NewClient(md.client)
+	cr, err := c.Read(context.Background(), storeId)
+	if err != nil {
+		t.Fatal(fmt.Errorf("got an error reading %q: %w", storeId, err))
+	}
+
+	// update Vault server to existing store
```
Comment (johanbrandhorst) [inline internal/provider/resource_credential_store_static_test.go:105]: I think this comment is probably out of place now :D

File: internal/provider/resource_scope_test.go [hunk 3]
```diff
@@ -42,6 +42,7 @@ resource "boundary_scope" "proj1" {
 	scope_id    = boundary_scope.org1.id
 	description = "foo"
 	depends_on = [boundary_role.org1_admin]
+	auto_create_admin_role = true // this is necessary for creating static credential store. without it, boundary returns a 403 on creation
```
Comment (johanbrandhorst) [inline internal/provider/resource_scope_test.go:45]: Would be interested in hearing Louis' thoughts on this

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

**Review by johanbrandhorst — APPROVED** [review]

File: internal/provider/resource_credential_store_static_test.go [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+	staticCredStoreNamespace = "static"
+	staticCredStoreUpdate    = "_random"
+)
+
+func staticCredStoreResource(name string, description string) string {
+	return fmt.Sprintf(`
+resource "boundary_credential_store_static" "example" {
+	name  = "%s"
+	description = "%s"
+	scope_id = boundary_scope.proj1.id
```
Comment (louisruch) [inline internal/provider/resource_credential_store_static_test.go:30]: Add `depends_on  = [boundary_role.proj1_admin]` to the resource to ensure the role exists before the cred store resource is created

File: internal/provider/resource_scope_test.go (hunk 3 above)
Comment (louisruch) [reply internal/provider/resource_scope_test.go:45]: I think this has to do with not setting the `depends_on` in the static resource, would be curious if that resolves it.

**Review by louisruch — COMMENTED** [review]
I believe adding the `depends_on` will remove the scope changes you added. Otherwise this looks perfect! Well done

**Review by johanbrandhorst — APPROVED** [review]
Lets ship it!

**Review by louisruch — APPROVED** [review]
LGTM :shipit:

**Pull request merged**